devel/build

```diff
@@ -19,19 +19,33 @@ export GIT_REVISION=$(git describe --tags --abbrev=40 --always --dirty || true)
 # Calling `docker run nextstrain/base` will still only pull down the small base
 # image rather than pulling down the larger nextstrain/base-builder image.
 
-docker build \
+platform=linux/amd64
+
+# `buildx create` is necessary to use a driver that supports multi-platform
+# images.
+builder=$(docker buildx create --driver docker-container)
+
+trap "docker buildx rm $builder" EXIT
+
+docker buildx build \
+    --builder "$builder" \
+    --platform $platform \
     --build-arg CACHE_DATE \
     --build-arg GIT_REVISION \
     --cache-from nextstrain/base-builder \
     --cache-from nextstrain/base \
+    --cache-to type=inline \
```

Comment:
> @tsibley: Running this locally a few times, I noticed that no caching happens. I think this is because the `--cache-from` statements in the context of `buildx` only look at the registry and not the local image cache (e.g. what `docker image ls` shows a view of). In CI this won't matter as much, as we don't expect the local image cache to be populated, but it makes the local dev loop quite a bit slower. I don't think we can make that better with BuildKit unless we cache to/from a local dir as well as the registry.

How did you notice this? I just ran this twice locally, and noticed the second time was near-instant with everything marked as `CACHED`:

```sh
export GIT_REVISION=$(git describe --tags --abbrev=40 --always --dirty || true)
export CACHE_DATE=$(date --utc +%Y%m%dT%H%M%SZ)
platform=linux/amd64
builder=$(docker buildx create --driver docker-container)
trap "docker buildx rm $builder" EXIT
time docker buildx build \
    --builder "$builder" \
    --platform $platform \
    --build-arg CACHE_DATE \
    --build-arg GIT_REVISION \
    --cache-from nextstrain/base-builder \
    --cache-from nextstrain/base \
    --cache-to type=inline \
    --tag nextstrain/base-builder \
    --target builder \
    .
```

Reply:
Hmm, I noticed it when running `./devel/build` in succession… though maybe not _immediate_ succession since it takes a few minutes and I did some other review work while it built. It's possible I perturbed something that invalidated the cache... it started from the top again IIRC. Trying again now…

Reply:
In case it's relevant while I wait, versions of things.

<details>

```
$ docker version
Client:
 Version:           20.10.7
 API version:       1.41
 Go version:        go1.13.8
 Git commit:        20.10.7-0ubuntu5~18.04.3
 Built:             Mon Nov  1 01:04:14 2021
 OS/Arch:           linux/amd64
 Context:           default
 Experimental:      true

Server:
 Engine:
  Version:          20.10.7
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.13.8
  Git commit:       20.10.7-0ubuntu5~18.04.3
  Built:            Fri Oct 22 00:57:37 2021
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.5.5-0ubuntu3~18.04.2
  GitCommit:        
 runc:
  Version:          1.0.1-0ubuntu2~18.04.1
  GitCommit:        
 docker-init:
  Version:          0.19.0
  GitCommit:        

$ docker buildx version
github.com/docker/buildx v0.7.1 05846896d149da05f3d6fd1e7770da187b52a247
```

</details>

Reply:
Two subsequent runs of `./devel/build` with no intervening changes both started from the top for me (i.e. with `apt-get update`).

In your test, did the same underlying builder get reused? (perhaps because the exit trap never fired if the same shell was used?)

I'm testing out sharing the builder now…

Reply:
Yep, that did it. The build cache is specific to the builder, which is unsurprising once you realize it.

See 6ae4036 for a fixup to re-use the builder between `./devel/build` invocations. This won't matter in CI, but helps a lot for local builds.

Reply:
Oh! Unsurprising but took two to catch 😅